### PR TITLE
Fix Jetpack blog detection

### LINKED ISSUE #4480: Analytics isn't counting Jetpack sites correctly
In [WPAnalyticsTrackerAutomatticTracks.m](https://github.com/wordpress-mobile/WordPress-iOS/blob/233c8baeb9753741efcba92fcd317e95c4144da2/WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m#L99), we're checking if we're using Jetpack like this:

```objc
jetpackBlogsPresent = [account jetpackBlogs].count > 0;
```

We have similar code in `WPAnalyticsTrackerMixpanel`.
I believe `jetpackBlogs` shouldn't be populated anymore for the default account

A more appropriate check (needs testing):

```objc
jetpackBlogsPresent = [[[blogService blogsForAllAccounts] wp_filter:^BOOL(Blog *blog) {
    return !blog.isHostedAtWPcom || blog.jetpackAccount != nil;
}] count] > 0;
```

I'm also considering if we should just drop `jetpackAccount`/`jetpackBlogs` now that Jetpack REST seems stable

cc @sendhil @beaulebens

### PR DESCRIPTION
**Pull request by koke**
The old code would only work for the pre-REST Jetpack sites connected to
the default account.

It now takes into account managed Jetpack sites, sites connected to a
different account, and sites that have Jetpack installed but not
configured in the app

Fixes #4480 

Needs Review: @sendhil

### CODE REVIEW

**Comment by koke** [comment]
Maybe a good candidate for 5.8?

**Comment by sendhil** [comment]
@koke  sorry I missed this, was out most of last week for Thanksgiving. Will look soon.

File: WordPress/Classes/Services/BlogService.m [hunk 1]
```diff
@@ -343,6 +343,22 @@ - (BOOL)hasVisibleWPComAccounts
...
+{
+    NSPredicate *jetpackManagedPredicate = [NSPredicate predicateWithFormat:@"account != NULL AND isHostedAtWPcom = NO"];
+    NSInteger jetpackManagedCount = [self blogCountWithPredicate:jetpackManagedPredicate];
+    if (jetpackManagedCount > 0) {
+        return YES;
+    }
+
+    NSArray *selfHostedBlogs = [self blogsWithNoAccount];
+    NSArray *jetpackUnmanagedBlogs = [selfHostedBlogs wp_filter:^BOOL(Blog *blog) {
+        return blog.jetpack.isConnected;
```
Comment (sendhil) [inline WordPress/Classes/Services/BlogService.m]: 😻

**Comment by sendhil** [comment]
Looks good. :shipit:

**Comment by sendhil** [comment]
@koke  going to cherry pick the commits into `release/5.8`

**Pull request merged**